app/src/main/java/com/example/parkingsystem/mvp/model/ParkingReservationModel.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.parkingsystem.mvp.model
+
+import com.example.parkingsystem.mvp.contract.ParkingReservationContract
+import com.example.parkingsystem.util.DateUtils
+import java.text.SimpleDateFormat
+import java.util.*
+
+class ParkingReservationModel : ParkingReservationContract.ParkingReservationModel {
+
+    private var isDateBeginPressed = false
+    private lateinit var dateBegin: Calendar
+    private lateinit var timeBegin: Calendar
+    private lateinit var dateEnd: Calendar
+    private lateinit var timeEnd: Calendar
+
+    override fun setDateBeginPressed(isDateBeginPressed: Boolean) {
+        this.isDateBeginPressed = isDateBeginPressed
+    }
+
+    override fun getDateBeginPressed(): Boolean = isDateBeginPressed
+
+    override fun convertToCalendar(sDate: String, formatGiven: String): Calendar {
+        val format = SimpleDateFormat(formatGiven, Locale.getDefault())
+        return DateUtils.convertToCalendar(sDate, format)
+    }
+
+    override fun getDateBegin(): Calendar = dateBegin
+
+    override fun getTimeBegin(): Calendar = timeBegin
+
+    override fun getDateEnd(): Calendar = dateEnd
+
+    override fun getTimeEnd(): Calendar = timeEnd
+
+    override fun setDateBegin(dateBegin: Calendar) {
+        this.dateBegin = dateBegin
+    }
+
+    override fun setTimeBegin(timeBegin: Calendar) {
+        this.timeBegin = timeBegin
+    }
+
+    override fun setDateEnd(dateEnd: Calendar) {
+        this.dateEnd = dateEnd
+    }
+
+    override fun setTimeEnd(timeEnd: Calendar) {
+        this.timeEnd = timeEnd
+    }
+
+    override fun getFormattedString(calendar: Calendar, formatGiven: String): String {
+        val format = SimpleDateFormat(formatGiven, Locale.getDefault())
+        return DateUtils.convertToString(calendar, format)
+    }
```

Comment:
please use = instead of return, and avoid single-use variables

Reply:
It doesn't allow me, but I erased the single use variable